schema/draft/schema.ts

```diff
@@ -1383,6 +1392,148 @@ export interface ElicitResult extends Result {
   content?: { [key: string]: string | number | boolean };
 }
 
+export interface Stream {
+  /**
+   * A unique identifier for the stream. If the stream is resumable, this ID
+   * should be globally unique across instances of the server.
+   */
+  streamId: string;
+
+  resumeInterval?: {
+    /**
+     * The minimum number of seconds a client should wait before resuming
+     * the stream after reconnection or between `stream/poll` requests.
+     */
+    min?: number;
+
+    /**
+     * The minimum number of seconds a client should wait before resuming
+     * the stream after reconnection. A value of 0 indicates that the stream
+     * is not resumable, and that the work will be cancelled upon disconnect.
+     */
+    max?: number;
+  }
+}
+
+/**
+ * A notification to the client that a stream was created. Messages may be sent
+ * on the stream immediately after the creation notification.
+ */
+export interface StreamCreateNotification extends Notification {
+  method: "notifications/stream/create";
+  params: {
+    /**
+     * The stream that was created.
+     */
+    stream: Stream;
+  }
+}
```

Comment:
Since `Stream` no longer extends `BaseMetadata`, what are your thoughts about inlining `streamId` and `resumeInterval` here?

Also, we may need a `requestId` property here so that a client can associate the stream with an originating request on multiplexed connections (e.g. stdio).

Reply:
Hmm, in the original design there was no `stream/poll/all`, and it was assumed that clients would track stream IDs internally.  If a client cancelled a tool call, it would need to know which stream to *not* resume or poll.

Perhaps with the addition of `stream/poll/all` that knowledge is no longer necessary.

Reply:
I would track stream IDs internally, but on reconnection `poll/all` to find what streams I need to resubscribe to. From a client POV streams are super lightweight and there's no reason I wouldn't just want to get all the data all the time.

Reply:
Another scenario in which the client must be certain of the request ID / stream ID mapping is webhooks.  In the original concept, stream IDs were not considered sensitive information, so a stream ID could be sent to a webhook in order to ping the client (see https://github.com/modelcontextprotocol/modelcontextprotocol/discussions/543#discussioncomment-13334781).  However, we are now considering stream IDs as sensitive information (which I think makes sense), so the server would need to send something else to the webhook instead.  Namely, the request ID.

I just pushed a commit that adds a `requestId` property to `notifications/stream/create` (and addresses some of the comments in this thread), but I'm open to removing it in favor of something better.